src/models/converters/balances.rs

```diff
@@ -2,11 +2,30 @@ use crate::models::backend::balances::Balance as BalanceDto;
 use crate::models::backend::chains::NativeCurrency;
 use crate::models::service::balances::Balance;
 use crate::providers::info::{TokenInfo, TokenType};
+use num_traits::{FromPrimitive, Pow};
 
 impl BalanceDto {
-    pub fn to_balance(&self, usd_to_fiat: f64, native_coin: &NativeCurrency) -> Balance {
-        let fiat_conversion = self.fiat_conversion.parse::<f64>().unwrap_or(0.0) * usd_to_fiat;
-        let fiat_balance = self.fiat_balance.parse::<f64>().unwrap_or(0.0) * usd_to_fiat;
+    pub fn to_balance(
+        &self,
+        token_to_usd: f64,
+        usd_to_fiat: f64,
+        native_coin: &NativeCurrency,
+    ) -> Balance {
+        let token_decimals = self
+            .token
+            .as_ref()
+            .and_then(|token| Some(token.decimals))
+            .unwrap_or(native_coin.decimals);
+
+        // We try to convert the token_decimals to f64, if it fails we fallback to the native currency decimals (f64), if it fails the function panics
+        let decimal_exp =
+            f64::from_u64(token_decimals).unwrap_or(f64::from_u64(native_coin.decimals).unwrap());
+
+        let token_balance =
```

Comment:
Is this datatype large enough? Is there something like BigInteger or BigNumber in Rust?

Reply:
Regarding Big Numbers in Rust I found the following discussion with some suggestions: https://www.reddit.com/r/rust/comments/a7frqj/have_anyone_reviewed_any_of_the_decimal_crates/

I've decided to go with the the 64 bit version already present for development but we can discuss about possible implementations for representing this value 👍